mesonbuild/build.py

```diff
@@ -1369,7 +1369,7 @@ def process_kwargs(self, kwargs):
         if 'depends' in kwargs:
             depends = unholder(listify(kwargs['depends']))
             for d in depends:
-                if not isinstance(d, BuildTarget):
+                if not (isinstance(d, BuildTarget, CustomTarget)):
```

Comment:
You put the first parenthesis at the wrong place; I recommended `not isinstance(d, (BuildTarget, CustomTarget))`